core/Metrics.php

```diff
@@ -242,6 +242,21 @@ public static function getUnit($column, $idSite)
             '_time_'  => 's'
         );
 
+        $unit = null;
+
+        /**
+         * Use this event to define units for custom metrics used in evolution graphs and row evolution only.
+         *
+         * @param string $unit should hold the unit (e.g. %, €, s or empty string)
+         * @param string $column name of the column to determine
+         * @param string $idSite id of the current site
+         */
+        Piwik::postEvent('Metrics.getEvolutionUnit', [&$unit, $column, $idSite]);
+
+        if (!empty($unit)) {
+            return $unit;
+        }
+
```

Comment:
Could this method get called a lot? If so might make sense to cache the value in the transient cache to avoid posting the event too many times.

Reply:
It's called in `JqplotDataGenerator`. Not sure if it's worth to cache that, as cache would need to be based on columnname and idsite. So likely a lot entries would be created